internal/providers/manager/manager.go

```diff
@@ -32,6 +33,8 @@ import (
 
 // ProviderManager encapsulates operations for manipulating Provider instances
 type ProviderManager interface {
+	CreateFromConfig(ctx context.Context, providerClass db.ProviderClass, projectID uuid.UUID, name string, config json.RawMessage) (*db.Provider, error)
+	CreateFromSessionState(ctx context.Context, providerClass db.ProviderClass, creds v1.Credential, state string) (*db.Provider, error)
```

Comment:
PTAL at this change to the providerManager - does it look OK?
The thinking was - we have two primary ways of createing providers, one is where the caller provides attributes like name or the config through the call parameters, typically when creating a provider with a token and the other is where the caller first stores the attributes in a state and the state is then retrieved during the oauth callback.